### PR TITLE
Eliminate ANRs

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/9388

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/9388) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Like in original issue
1. Run debug build
2. Go through messages like StrictMode policy violation in device monitor

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
CI results:
Android, Linux - succeeded;
iOS, macOS - failed; 
Windows - aborted.

Marking Android and Linux to skip.
Deleting b-b branch, rebasing  b-c branch and restarting.

**Comment by AlexeyBarabash** [comment]
b-b branch was up-to-date in fact, so just restarting build.

**Comment by SergeyZhukovsky** [comment]
I would uplift it to 1.8 probably as it's something we want to fix ASAP.

File: android/java/org/chromium/chrome/browser/BraveActivity.java [hunk 1]
```diff
@@ -239,13 +241,18 @@ private void createNotificationChannel() {
...
-        Context context = ContextUtils.getApplicationContext();
-        if (BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(this)) {
-            // Don't ask again
-            return;
-        }
-        Intent intent = new Intent(context, BraveSetDefaultBrowserNotificationService.class);
-        context.sendBroadcast(intent);
+        // Post task to IO thread because isBraveSetAsDefaultBrowser may cause
+        // sqlite file IO operation underneath
+        PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK, () ->
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/BraveActivity.java:246]: this is nice thing to know.

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/BraveActivity.java:246]: Yeah, it was a surprise for me. I saw it on Samsung Android 8 device but didn't on Xiaomi Android 9 device.
here is a part of logs from Samsung 
```
04-22 15:40:02.737: D/StrictMode(11681): StrictMode policy violation; ~duration=144 ms: android.os.StrictMode$StrictModeDiskReadViolation: policy=22085695 violation=2
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1438)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteConnection.applyBlockGuardPolicy(SQLiteConnection.java:1247)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteConnection.executeForCursorWindow(SQLiteConnection.java:1014)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:836)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:152)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:141)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:219)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:258)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.storage.EdmStorageProviderBase.getValues(EdmStorageProviderBase.java:389)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.queryAllDefaultAppIntents(ApplicationPolicy.java:12763)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.retrieveDefaultAppFromDb(ApplicationPolicy.java:12802)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.getDefaultApplicationInternal(ApplicationPolicy.java:12109)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.filterCandidatesWithDomainPreferredActivitiesLPr(PackageManagerService.java:10180)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.queryIntentActivitiesInternal(PackageManagerService.java:9838)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.resolveIntentInternal(PackageManagerService.java:8772)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.resolveIntent(PackageManagerService.java:8745)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:930)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:5096)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Binder.execTransact(Binder.java:682)
04-22 15:40:02.737: D/StrictMode(11681): # via Binder call with stack:
04-22 15:40:02.737: D/StrictMode(11681): android.os.StrictMode$LogStackTrace
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.StrictMode.readAndHandleBinderCallViolations(StrictMode.java:2137)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Parcel.readExceptionCode(Parcel.java:1941)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Parcel.readException(Parcel.java:1910)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.content.pm.IPackageManager$Stub$Proxy.resolveIntent(IPackageManager.java:4554)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ApplicationPackageManager.resolveActivityAsUser(ApplicationPackageManager.java:1035)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ApplicationPackageManager.resolveActivity(ApplicationPackageManager.java:1029)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.notifications.BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(BraveSetDefaultBrowserNotificationService.
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.BraveActivity.setupBraveSetDefaultBrowserNotification(BraveActivity.java:249)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.BraveActivity.performPostInflationStartup(BraveActivity.java:167)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeTabbedActivity.performPostInflationStartup(ChromeTabbedActivity.java:1547)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.postInflationStartup(AsyncInitializationActivity.java:182)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.onInflationComplete(ChromeBrowserInitializer.java:176)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.lambda$handlePreNativeStartup$0$ChromeBrowserInitializer(ChromeBrowserInitializer.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer$$Lambda$0.run(ChromeBrowserInitializer.java:0)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onInitialLayoutInflationComplete(AsyncInitializationActivity.java:753)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.onInitialLayoutInflationComplete(ChromeActivity.java:596)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.doLayoutInflation(ChromeActivity.java:568)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.triggerLayoutInflation(ChromeActivity.java:522)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.setContentViewAndLoadLibrary(AsyncInitializationActivity.java:168)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.handlePreNativeStartup(ChromeBrowserInitializer.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onCreateInternal(AsyncInitializationActivity.java:318)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onCreate(AsyncInitializationActivity.java:264)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.Activity.performCreate(Activity.java:7183)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1220)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2910)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3032)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.-wrap11(ActivityThread.java:0)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Handler.dispatchMessage(Handler.java:105)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Looper.loop(Looper.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.main(ActivityThread.java:6940)
04-22 15:40:02.737: D/StrictMode(11681): <----->at java.lang.reflect.Method.invoke(Method.java)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)

```

**Comment by AlexeyBarabash** [comment]
CI for iOS and Windows succeeded.
macOS failed for 
`test-unit`
```
22:58:16  INFO: Starting to record.
22:58:16  INFO: Processing GoogleTest-1.8
22:58:26  INFO: [GoogleTest-1.8] - 1 test report file(s) were found with the pattern 'src/brave_unit_tests.xml' relative to '/Users/jenkins/jenkins/workspace/ve-browser-build-pr_android_anrs' for the testing framework 'GoogleTest-1.8'.
```
and 
`test-browser`
```
23:08:27  INFO: Starting to record.
23:08:27  INFO: Processing GoogleTest-1.8
23:08:27  INFO: [GoogleTest-1.8] - 1 test report file(s) were found with the pattern 'src/brave_browser_tests.xml' relative to '/Users/jenkins/jenkins/workspace/ve-browser-build-pr_android_anrs' for the testing framework 'GoogleTest-1.8'.
```
Restarting

**Comment by AlexeyBarabash** [comment]
Verified Nightly with https://github.com/brave/brave-browser/releases/tag/v1.10.9 - have no crashes at start.
Verified with modified Nightly (master sources) I can setup default browser through the notification, I cann't see strict mode violation in monitor.

The patch used is 
```
git diff
diff --git a/android/java/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java b/android/java/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java
index 752214f6b..ffe5b1e5a 100644
--- a/android/java/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java
+++ b/android/java/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java
@@ -164,6 +164,8 @@ public class BraveSetDefaultBrowserNotificationService extends BroadcastReceiver
         currentTime.set(Calendar.MINUTE, NOTIFICATION_MIN);
         currentTime.set(Calendar.SECOND, 0);
         currentTime.set(Calendar.MILLISECOND, 0);
+currentTime = Calendar.getInstance();
+currentTime.add(Calendar.SECOND, 30);
         AlarmManager am = (AlarmManager) mContext.getSystemService(Context.ALARM_SERVICE);
         Intent intent = new Intent(mContext, BraveSetDefaultBrowserNotificationService.class);
         intent.putExtra(INTENT_TYPE, AT_11_22);
diff --git a/android/java/org/chromium/chrome/browser/util/BraveReferrer.java b/android/java/org/chromium/chrome/browser/util/BraveReferrer.java
index d531dccb9..dd49a0fea 100644
--- a/android/java/org/chromium/chrome/browser/util/BraveReferrer.java
+++ b/android/java/org/chromium/chrome/browser/util/BraveReferrer.java
@@ -109,6 +109,7 @@ public class BraveReferrer implements InstallReferrerStateListener {
                     Uri uri = Uri.parse("http://www.stub.co/?"+referrer);
                     // Get and save user referal program code
                     String urpc = uri.getQueryParameter("urpc");
+urpc = "urpc123";
                     if (urpc != null && !urpc.isEmpty()) {
                         PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK,
                             new SaveReferrerRunnable(urpc));
```

**Pull request merged**